### ISSUE TITLE
[BUG] CSI VolumeSnapshot fails to restore if the source PV/PVC is in detached status

### ISSUE THREAD

**Issue by ddalton** [longhorn/longhorn#4834]
## Describe the bug

Run a job that creates a PVC and once the job completes the PVC goes into detached status.
Before the job completes it takes a snapshot of the PVC.
Later on during restore when PVC is created referring to this snapshot it fails with the following error:

```
Events:
  Type     Reason                Age                    From                                                                                     Message
  ----     ------                ----                   ----                                                                                     -------
  Warning  ProvisioningFailed    60m (x3 over 78m)      driver.longhorn.io_csi-provisioner-d7df997cf-lg8dr_b777a221-95bc-4ce9-9cdd-eb60d449dae3  failed to provision volume with StorageClass "longhorn": rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [detail=, message=unable to create volume: unable to create volume pvc-fff51adc-95a8-4688-97ad-40e7b5b9cde1: failed to verify data source: cannot get client for volume pvc-a6a813a6-5d17-4ab1-9fcb-999b2297ed37: engine is not running, code=Server Error] from [http://longhorn-backend:9500/v1/volumes]
  Warning  ProvisioningFailed    50m (x8 over 78m)      driver.longhorn.io_csi-provisioner-d7df997cf-lg8dr_b777a221-95bc-4ce9-9cdd-eb60d449dae3  failed to provision volume with StorageClass "longhorn": rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=Server Error, detail=, message=unable to create volume: unable to create volume pvc-fff51adc-95a8-4688-97ad-40e7b5b9cde1: failed to verify data source: cannot get client for volume pvc-a6a813a6-5d17-4ab1-9fcb-999b2297ed37: engine is not running] from [http://longhorn-backend:9500/v1/volumes]
  Normal   ExternalProvisioning  2m58s (x302 over 78m)  persistentvolume-controller                                                              waiting for a volume to be created, either by external provisioner "driver.longhorn.io" or manually created by system administrator
  Normal   Provisioning          2m8s (x28 over 78m)    driver.longhorn.io_csi-provisioner-d7df997cf-lg8dr_b777a221-95bc-4ce9-9cdd-eb60d449dae3  External provisioner is provisioning volume for claim "cdp-drs/restore-job-cbf8f827-5c9b-4e8d-9b14-21550d99b1ae"
```

## To Reproduce

1. Create a Kubernetes job that saves some data to a PVC
2. Ensure the cluster has a VolumeSnapshotClass configured with type: snap
```
apiVersion: snapshot.storage.k8s.io/v1
deletionPolicy: Delete
driver: driver.longhorn.io
kind: VolumeSnapshotClass
metadata:
  name: longhorn
parameters:
  type: snap
```
4. Just before the job completed take a snapshot named 'drs-k8s-object-storage-dc34e43c-bc40-4850-b544-cadd7cd67704' of the PVC
5. Ensure the READYTOUSE value is true
6. Create a new PVC referring to the volume snapshot
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: restore-job-cbf8f827-5c9b-4e8d-9b14-21550d99b1ae
  namespace: cdp-drs
spec:
  accessModes:
  - ReadWriteOnce
  dataSource:
    apiGroup: snapshot.storage.k8s.io
    kind: VolumeSnapshot
    name: drs-k8s-object-storage-dc34e43c-bc40-4850-b544-cadd7cd67704
  resources:
    requests:
      storage: 100Mi
  storageClassName: longhorn
  volumeMode: Filesystem
status:
  phase: Pending
```

7. The new PVC is in Pending phase and the errors are listed in the events section of the PVC. 

## Expected behavior

The restore PVC should be created successfully.
On a side node, this functionality works fine in OpenShift that uses a Ceph storage provider.

## Log or Support bundle
[longhorn_support_bundle.zip](https://github.com/longhorn/longhorn/files/9941151/longhorn_support_bundle.zip)

## Environment

 - Longhorn version: 1.3.2
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE2 v1.23.12
   - Number of management node in the cluster: 1
   - Number of worker node in the cluster: 1
 - Node config
   - OS type and version:
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): AWS
 - Number of Longhorn volumes in the cluster:

**Comment by ddalton** [comment]
Thanks @PhanLe1010. 
Would this fix also address snapshots of deleted PVCs?

**Comment by innobead** [comment]
This has been fixed by introducing #3715  which was introduced in v1.5.0 .
